### DISCUSSION TITLE
Help with create queries

### DISCUSSION THREAD

**Discussion by karenarcoverde** [Blacksuan19/redash-python#7] [Q&A] [answered]
Hi. I need your help. I'm confuse about the function create in Class QueriesService. I need to put the argument in a Dict.  def create(data: Dict) -> Dict
but what the arguments in the Dict? for example, I need to put name, data_source_id... what are the exact arguments?

Thanks!

**Comment by Blacksuan19** [comment, marked as answer]
this method is designed to be used with another query you have already fetched from redash and modified, for example

```python
from redash_python import Redash

rd = Redash(base_url="", api_key="")

# fetch an already existing query
que = rd.queries.get(1)

# modify query data
que["name"] = "new name"
que["tags"] = ["tag1", "tag2"]
que["query"] = "select column1, column2 from table"

# create new query
rd.queries.create(que)  
```

you can get the structure of a query by printing it `print(rd.queries.get(1))`
